src/sql.rs

```diff
@@ -47,7 +92,7 @@ impl Sql {
     pub fn open(&self, context: &Context, dbfile: &std::path::Path, readonly: bool) -> bool {
         match open(context, self, dbfile, readonly) {
             Ok(_) => true,
-            Err(Error::SqlAlreadyOpen) => false,
+            Err(crate::error::Error::SqlError(Error::SqlAlreadyOpen)) => false,
```

Comment:
does it make sense to use CrateError as an alias as crate::error::Error everywhere?

Reply:
+1, why not

Reply:
Eventually this `crate::error::Error` shouldn't be needed at all. If the function returns `SqlError`, it has to be in the `sql` module. If it is inside some other module, it should be wrapped into its own variant.

In this case the function should return `SqlError` or its own error eventually and `error::Error` will get removed.